client/web/src/global/CoolCodeIntel.tsx

```diff
@@ -221,17 +298,45 @@ export const ReferencesList: React.FunctionComponent<
                 className={classNames('py-0 my-0', styles.referencesFilter)}
                 type="text"
                 placeholder="Filter by filename..."
-                value={filter === undefined ? '' : filter}
+                value={debouncedFilter === undefined ? '' : debouncedFilter}
```

Comment:
```suggestion
                value={filter === undefined ? '' : filter}
```

We want the input UI to be as responsive as possible, but the actual network request should be the one that is debounced. 

Also, I think we should decouple this from the loading/error/!data states. The problem here will be that the new filter value will trigger a new request, render a loading spinner and unmount this input, which will make the element lose focus and maybe lead to other bugs. Easiest solution would probably be to hoist this out of `ReferencesList` and manage the `Input` in the parent component